File: scraper/api_client.py
import httpx
import logging

class APIClient:
    def __init__(self, base_url, api_key):
        self.base_url = base_url.rstrip("/")
        self.headers = {
            "X-API-KEY": api_key,
            "Content-Type": "application/json"
        }

    async def post_job(self, job_data):
        url = f"{self.base_url}/api/scraper/jobs"
        async with httpx.AsyncClient() as client:
            try:
                response = await client.post(url, json=job_data, headers=self.headers, timeout=30.0)
                response.raise_for_status()
                return response.json()
            except Exception as e:
                logging.error(f"Failed to post job: {e}")
                return None

    async def post_alert(self, message, alert_type="system"):
        url = f"{self.base_url}/api/scraper/alert"
        payload = {"message": message, "type": alert_type}
        async with httpx.AsyncClient() as client:
            try:
                response = await client.post(url, json=payload, headers=self.headers, timeout=10.0)
                response.raise_for_status()
                return response.json()
            except Exception as e:
                logging.error(f"Failed to post alert: {e}")
                return None

File: scraper/utils/api_client.js
const axios = require('axios');
require('dotenv').config();

class APIClient {
    constructor() {
        this.apiUrl = process.env.BACKEND_API_URL || 'http://localhost:3000/api/scraper';
        this.apiKey = process.env.SCRAPER_API_KEY || 'dev_scraper_key';
    }

    async postJob(jobData) {
        try {
            const response = await axios.post(`${this.apiUrl}/jobs`, jobData, {
                headers: {
                    'x-api-key': this.apiKey,
                    'Content-Type': 'application/json'
                }
            });
            return response.data;
        } catch (error) {
            console.error(`API Post error for ${jobData.title}:`, error.response?.data || error.message);
            throw error;
        }
    }

    async sendSystemAlert(message) {
        try {
            await axios.post(`${this.apiUrl}/alert`, {
                message: message,
                type: 'system'
            }, {
                headers: {
                    'x-api-key': this.apiKey
                }
            });
        } catch (error) {
            console.error('Failed to send system alert:', error.message);
        }
    }
}

module.exports = new APIClient();
